src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -250,10 +250,10 @@ class let:
     """
 
     @typing.overload
-    def __init__(self, var: str | itir.Sym, init_form: itir.Expr): ...
+    def __init__(self, var: str | itir.Sym, init_form: itir.Expr | str): ...
 
     @typing.overload
-    def __init__(self, *args: Iterable[tuple[str | itir.Sym, itir.Expr]]): ...
+    def __init__(self, *args: Iterable[tuple[str | itir.Sym, itir.Expr | str]]): ...
 
     def __init__(self, *args):
```

Comment:
does type checking work properly if the implementation is not properly annotated with the union of the above?

Reply:
Likely no one ever used `let(sym_a, sym_b)(...)` with `sym_b` being a string (as such rebinding cases usually mean `sym_b` is already an `itir.Sym` coming from somewhere). I took your suggestion.

Reply:
You mean not annotating?